exporter/opentelemetry-exporter-otlp-proto-grpc/src/opentelemetry/exporter/otlp/proto/grpc/_metric_exporter/__init__.py

```diff
@@ -79,90 +80,124 @@ def __init__(
         )
 
     def _translate_data(
-        self, data: Sequence[Metric]
+        self, data: MetricsData
     ) -> ExportMetricsServiceRequest:
-        sdk_resource_scope_metrics = {}
 
-        for metric in data:
-            resource = metric.resource
-            scope_map = sdk_resource_scope_metrics.get(resource, {})
-            if not scope_map:
-                sdk_resource_scope_metrics[resource] = scope_map
+        resource_instrumentation_scope_pb2_scope_metrics = {}
 
-            scope_metrics = scope_map.get(metric.instrumentation_scope)
+        for resource_metrics in data.resource_metrics:
 
-            if not scope_metrics:
-                if metric.instrumentation_scope is not None:
-                    scope_map[metric.instrumentation_scope] = pb2.ScopeMetrics(
+            resource = resource_metrics.resource
+
+            instrumentation_scope_pb2_scope_metrics = (
+                resource_instrumentation_scope_pb2_scope_metrics.get(
+                    resource, {}
+                )
+            )
+
+            if not instrumentation_scope_pb2_scope_metrics:
+                resource_instrumentation_scope_pb2_scope_metrics[
+                    resource
+                ] = instrumentation_scope_pb2_scope_metrics
+
+            for scope_metrics in resource_metrics.scope_metrics:
+
+                instrumentation_scope = scope_metrics.scope
+
+                if instrumentation_scope is None:
+                    instrumentation_scope_pb2_scope_metrics[
+                        None
+                    ] = pb2.ScopeMetrics()
+
+                else:
+                    instrumentation_scope_pb2_scope_metrics[
+                        instrumentation_scope
+                    ] = pb2.ScopeMetrics(
```

Comment:
will this scope_metrics contain unique `scope` list items? Otherwise this replaces the prev added items?